src/overlays/actors/ovl_Bg_Spot01_Idohashira/z_bg_spot01_idohashira.c

```diff
@@ -21,49 +49,304 @@ const ActorInit Bg_Spot01_Idohashira_InitVars = {
+void func_808AAE6C(BgSpot01Idohashira* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Vec3f sp30 = this->dyna.actor.posRot.pos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idohashira/func_808AAF34.s")
+    sp30.y += kREG(15);
+    func_80033480(globalCtx, &sp30, kREG(11) + 350.0f, kREG(12) + 5, kREG(13) + 0x7D0, kREG(14) + 0x320, 0);
+    func_808AAD3C(globalCtx, &sp30, 5);
+    BgSpot01Idohashira_PlayBreakSfx2(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idohashira/BgSpot01Idohashira_Destroy.s")
+void func_808AAF34(BgSpot01Idohashira* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    Vec3f dest;
+    Vec3f src;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idohashira/func_808AB124.s")
+    if (this->unk_170 != 0) {
+        src.x = kREG(20) + 1300.0f;
+        src.y = kREG(21) + 200.0f;
+        src.z = 0.0f;
+        Matrix_MultVec3f(&src, &dest);
+        func_80033480(globalCtx, &dest, kREG(16) + 80.0f, kREG(17) + 10, kREG(18) + 1000, kREG(19), 0);
+        func_808AAD3C(globalCtx, &dest, 3);
+        src.x = -(kREG(20) + 1300.0f);
+        src.y = kREG(21) + 200.0f;
+        src.z = 0.0f;
+        Matrix_MultVec3f(&src, &dest);
+        func_80033480(globalCtx, &dest, kREG(16) + 80.0f, kREG(17) + 10, kREG(18) + 1000, kREG(19), 0);
+        func_808AAD3C(globalCtx, &dest, 3);
+        this->unk_170 = 0;
+        BgSpot01Idohashira_PlayBreakSfx1(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idohashira/func_808AB144.s")
+void BgSpot01Idohashira_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot01Idohashira* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idohashira/func_808AB18C.s")
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idohashira/func_808AB1DC.s")
+s32 BgSpot01Idohashira_NotInCsMode(GlobalContext* globalCtx) {
+    if (globalCtx->csCtx.state == 0) {
+        return 1;
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idohashira/func_808AB29C.s")
+CsCmdActorAction* BgSpot01Idohashira_GetNpcAction(GlobalContext* globalCtx, s32 actionIdx) {
+    s32 pad[2];
+    CsCmdActorAction* npcAction = NULL;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idohashira/func_808AB3E8.s")
+    if (!BgSpot01Idohashira_NotInCsMode(globalCtx)) {
+        npcAction = globalCtx->csCtx.npcActions[actionIdx];
+    }
+    return npcAction;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idohashira/func_808AB3F8.s")
+void func_808AB18C(BgSpot01Idohashira* this) {
+    this->dyna.actor.shape.rot.x += kREG(6);
+    this->dyna.actor.shape.rot.y += (s16)(kREG(7) + 0x3E8);
+    this->dyna.actor.shape.rot.z += (s16)(kREG(8) + 0x7D0);
```

Comment:
Are explicit casts needed here?
```suggestion
    this->dyna.actor.shape.rot.y += kREG(7) + 0x3E8;
    this->dyna.actor.shape.rot.z += kREG(8) + 0x7D0;
```

Reply:
I just double-checked, and yes, they're necessary unfortunately.